README
NAME
    Dist::Zilla::Plugin::Git::NextRelease - Use time-stamp from Git instead
    of process start time.

VERSION
    version 0.004001

SYNOPSIS
    This module acts as a moderately thin wrapper to "[NextRelease]" so that
    the time-stamps produced are generated by asking "git" for the
    time-stamp of the current branch.

    If you always require a specific branch for generating time-stamps, it
    can be specified as a parameter.

        -[NextRelease]
        +[Git::NextRelease]

    Optionally:

        +branch = master

    Or:

        +default_branch = master

    ( The latter only having impact in detached-head conditions )

    This exists mostly because of my extensive use of "[Git::CommitBuild]",
    to provide a commit series for both releases, and builds of all
    changes/commits in order to push them to Travis for testing. ( Mostly,
    because testing a build branch is substantially faster than testing a
    master that requires "Dist::Zilla", especially if you're doing "Fresh
    install" testing like I am. )

METHODS
  "section_header"
    This is the sole method of "[NextRelease]" that we override, in order
    for it to inject the right things.

    This method basically returns the date string to append to the Changes
    header.

ATTRIBUTES
  "branch"
    If set, always use the specified branch to determine time-stamp.

    Default value is resolved from determining "current" branch.

  "default_branch"
    If you want being on a branch to always resolve to that branch, but you
    still want a useful behavior when on a detached head, specifying this
    value means that on a detached head, the stated branch will be used
    instead.

FORMATS
    "[Git::NextRelease]" enhances and adds a few features of "[NextRelease]"

    *   %d - A CLDR formatter of the "branch"'s "HEAD" timestamp. ( Original
        uses simply "now" )

    *   %H - The SHA1 of "branch"'s "HEAD"

    *   %h - The SHA1 of "branch"'s "HEAD" shortened to 7 characters.

AUTHOR
    Kent Fredric <kentnl@cpan.org>

COPYRIGHT AND LICENSE
    This software is copyright (c) 2014 by Kent Fredric
    <kentfredric@gmail.com>.

    This is free software; you can redistribute it and/or modify it under
    the same terms as the Perl 5 programming language system itself.



Changes
Release history for Dist-Zilla-Plugin-Git-NextRelease

0.004002 2017-03-04T02:18:18Z c39216a

0.004001 2017-03-04T01:45:51Z 15d19c3
 [Bugfix]
 - Avoid test failures on 5.26 w/ -Ddefault_inc_excludes_dot
 - Closes https://github.com/kentnl/Dist-Zilla-Plugin-Git-NextRelease/issues/1

 [Dependencies::Stats]
 - Dependencies changed since 0.004000, see misc/*.deps* for details
 - configure: +1 (recommends: ↑1)
 - develop: +14 ↑4 -3 (recommends: +1, suggests: ↑2)
 - runtime: (recommends: +1)
 - test: +1 ↓1 (recommends: ↑1 ↓1)

0.004000 2014-10-04T01:35:02Z @8dca09d
 [Dependencies::Stats]
 - Dependencies changed since 0.003000, see misc/*.deps* for details
 - develop: (suggests: ↑1)
 - test: (recommends: ↑1)

 [Features]
 - Added new format tokens %H and %h that refer to SHA1 and SHA1-abbrev of branch's head commmit.

0.003000 2014-08-31T20:47:43Z
 [00 Minor]
 - Tests refactored
 - Add dump_config
 - metadata

 [Dependencies::Stats]
 - Dependencies changed since 0.002012, see misc/*.deps* for details
 - develop: +1 ↑1 (suggests: ↑2)
 - runtime: +1
 - test: +1 -4

 [Metadata]
 - set author = kentnl

 [Tests]
 - Refactored tests in terms of dztest.

0.002012 2014-08-15T21:16:31Z
 [00 Trivial]
 - CPANDAY!
 - no code changes
 - tests augmented
 - whitespace adjusted.

 [Dependencies::Stats]
 - Dependencies changed since 0.002011, see misc/*.deps* for details
 - configure: ↓1
 - develop: +1 (suggests: ↑1)
 - test: +1 -3 (recommends: ↑1 -1)

 [Misc]
 - Improve source side POD to reduce build side whitespace.

 [Tests]
 - update Test::ReportPrereqs

0.002011 2014-07-16T14:58:54Z
 [00 Trivial]
 - Misc packaging cleanups

 [Dependencies::Stats]
 - Dependencies changed since 0.002010, see misc/*.deps* for details
 - configure: (recommends: ↑1)
 - develop: +3 -3 (suggests: +2 -1)
 - test: (recommends: ↑2)

0.002010 2014-01-31T04:25:22Z
 [00 Minor]
 - Toolkit switch to EUMM.soft

 [Dependencies::AggregateChange]
 - 5.8 : 206 → 205
 - 5.10 : 185 → 184
 - 5.19 : 170 → 167

 [Dependencies::Stats]
 - Dependencies changed since 0.002001, see misc/*.deps* for details
 - build: -1
 - configure: +1 -1 (recommends: +1)
 - develop: +2 ↑1 -2
 - test: +1 ↓1 (recommends: +2 ↑1)

0.002001 2014-01-15T02:28:43Z
 [00 Bug Fix]
 - use newer dzil for encoded_content

 [Dependencies::Stats]
 - Dependencies changed since 0.002000, see misc/*.deps* for details
 - develop: +1 ↑1 -2
 - runtime: +1
 - test: +3 (recommends: +2)

0.002000 2014-01-13T22:29:38Z
 [00 Minor]
 - New attribute default_branch which will be used if you're on a detached head, and only on a detached head.

 [Dependencies::Stats]
 - Dependencies changed since 0.001001, see misc/*.deps* for details
 - develop: ↑1
 - test: +1 -1

0.001001 2014-01-12T21:05:29Z
 [Bugfix]
 - Bump to force a fixed Git::Wrapper::Plus with shallow repo fails.
 - Also to finish bootstapping

 [Dependencies::Stats]
 - Dependencies changed since 0.001000, see misc/*.deps* for details
 - develop: +1 ↑1 -1
 - runtime: +1 ↑2

0.001000 2014-01-12T01:21:35Z
 - First version.


Build.PL

# This file was automatically generated by Dist::Zilla::Plugin::ModuleBuild v5.011.
use strict;
use warnings;

use Module::Build 0.3601;


my %module_build_args = (
  "build_requires" => {
    "Module::Build" => "0.4204"
  },
  "configure_requires" => {
    "Module::Build" => "0.4204"
  },
  "dist_abstract" => "Use time-stamp from Git instead of process start time.",
  "dist_author" => [
    "Kent Fredric <kentfredric\@gmail.com>"
  ],
  "dist_name" => "Dist-Zilla-Plugin-Git-NextRelease",
  "dist_version" => "0.002001",
  "license" => "perl",
  "module_name" => "Dist::Zilla::Plugin::Git::NextRelease",
  "recommends" => {},
  "recursive_test_files" => 1,
  "requires" => {
    "DateTime" => 0,
    "Dist::Zilla" => "5.000",
    "Dist::Zilla::Plugin::NextRelease" => 0,
    "Git::Wrapper::Plus" => "0.003100",
    "Moose" => 0,
    "String::Formatter" => "0.100680",
    "perl" => "5.008",
    "strict" => 0,
    "utf8" => 0,
    "warnings" => 0
  },
  "script_files" => [],
  "test_requires" => {
    "Cwd" => 0,
    "ExtUtils::MakeMaker" => 0,
    "File::Copy::Recursive" => 0,
    "File::Spec::Functions" => 0,
    "FindBin" => 0,
    "Git::Wrapper::Plus::Support" => 0,
    "Git::Wrapper::Plus::Tester" => 0,
    "List::Util" => 0,
    "Path::Tiny" => 0,
    "Test::DZil" => 0,
    "Test::Fatal" => 0,
    "Test::More" => "1.001002"
  }
);


my %fallback_build_requires = (
  "Cwd" => 0,
  "ExtUtils::MakeMaker" => 0,
  "File::Copy::Recursive" => 0,
  "File::Spec::Functions" => 0,
  "FindBin" => 0,
  "Git::Wrapper::Plus::Support" => 0,
  "Git::Wrapper::Plus::Tester" => 0,
  "List::Util" => 0,
  "Module::Build" => "0.4204",
  "Path::Tiny" => 0,
  "Test::DZil" => 0,
  "Test::Fatal" => 0,
  "Test::More" => "1.001002"
);


unless ( eval { Module::Build->VERSION(0.4004) } ) {
  delete $module_build_args{test_requires};
  $module_build_args{build_requires} = \%fallback_build_requires;
}

my $build = Module::Build->new(%module_build_args);

$build->create_build_script;


maint/perlcritic.rc.gen.pl
#!/usr/bin/env perl
## no critic (Modules::RequireVersionVar)

# ABSTRACT: Write an INI file from a bundle

use 5.008;    # utf8
use strict;
use warnings;
use utf8;

our $VERSION = 0.001;

use Carp qw( croak carp );
use Perl::Critic::ProfileCompiler::Util qw( create_bundle );
use Path::Tiny qw(path);

## no critic (ErrorHandling::RequireUseOfExceptions)
my $bundle = create_bundle('Example::Author::KENTNL');
$bundle->configure;

my @stopwords = (
  qw(
  CLDR timestamp SHA formatter
    )
);
for my $wordlist (@stopwords) {
  $bundle->add_or_append_policy_field( 'Documentation::PodSpelling' => ( 'stop_words' => $wordlist ) );
}

$bundle->add_or_append_policy_field(
  'Subroutines::ProhibitCallsToUndeclaredSubs' => ( 'exempt_subs' => 'String::Formatter::_format_version' ), );

$bundle->remove_policy('ErrorHandling::RequireUseOfExceptions');

#$bundle->remove_policy('CodeLayout::RequireUseUTF8');

#$bundle->remove_policy('ErrorHandling::RequireCarping');
#$bundle->remove_policy('NamingConventions::Capitalization');

my $inf = $bundle->actionlist->get_inflated;

my $config = $inf->apply_config;

{
  my $rcfile = path('./perlcritic.rc')->openw_utf8;
  $rcfile->print( $config->as_ini, "\n" );
  close $rcfile or croak 'Something fubared closing perlcritic.rc';
}
my $deps = $inf->own_deps;
{
  my $target = path('./misc');
  $target->mkpath if not $target->is_dir;

  my $depsfile = $target->child('perlcritic.deps')->openw_utf8;
  for my $key ( sort keys %{$deps} ) {
    $depsfile->printf( "%s~%s\n", $key, $deps->{$key} );
    *STDERR->printf( "%s => %s\n", $key, $deps->{$key} );
  }
  close $depsfile or carp 'Something fubared closing perlcritic.deps';
}



Makefile.PL
# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v6.008.
use strict;
use warnings;

use 5.008;

use ExtUtils::MakeMaker;

my %WriteMakefileArgs = (
  "ABSTRACT" => "Use time-stamp from Git instead of process start time.",
  "AUTHOR" => "Kent Fredric <kentnl\@cpan.org>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0
  },
  "DISTNAME" => "Dist-Zilla-Plugin-Git-NextRelease",
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.008",
  "NAME" => "Dist::Zilla::Plugin::Git::NextRelease",
  "PREREQ_PM" => {
    "DateTime" => 0,
    "Dist::Zilla" => "5.000",
    "Dist::Zilla::Plugin::NextRelease" => 0,
    "Dist::Zilla::Util::ConfigDumper" => 0,
    "Git::Wrapper::Plus" => "0.003100",
    "Moose" => 0,
    "String::Formatter" => "0.100680",
    "strict" => 0,
    "utf8" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "Dist::Zilla::Util::Test::KENTNL" => "1.003001",
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "Git::Wrapper::Plus::Support" => 0,
    "Git::Wrapper::Plus::Tester" => 0,
    "Test::DZil" => 0,
    "Test::Fatal" => 0,
    "Test::More" => 0
  },
  "VERSION" => "0.004002",
  "test" => {
    "TESTS" => "t/*.t t/00-compile/*.t"
  }
);


my %FallbackPrereqs = (
  "DateTime" => 0,
  "Dist::Zilla" => "5.000",
  "Dist::Zilla::Plugin::NextRelease" => 0,
  "Dist::Zilla::Util::ConfigDumper" => 0,
  "Dist::Zilla::Util::Test::KENTNL" => "1.003001",
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "Git::Wrapper::Plus" => "0.003100",
  "Git::Wrapper::Plus::Support" => 0,
  "Git::Wrapper::Plus::Tester" => 0,
  "Moose" => 0,
  "String::Formatter" => "0.100680",
  "Test::DZil" => 0,
  "Test::Fatal" => 0,
  "Test::More" => 0,
  "strict" => 0,
  "utf8" => 0,
  "warnings" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);
